harness/propertyHelper.js

```diff
@@ -1,5 +1,6 @@
 
 function isConfigurable(obj, name) {
+    var desc = Object.getOwnPropertyDescriptor(obj, name);
```

Comment:
There's a fine distinction between the type of dangers from destructive
assertions. I agree with @bterlson about test hygiene, but I also recognize the
"footgun" described by @ljharb.

One approach that might address both concerns is by encapsulating all property
descriptor checks into a single assert method. It could still be destructive
(limiting complexity and promoting test hygiene), but it would abstract away
the gotchas of the common case (testing multiple descriptors on the same
object).

We last spoke about this in March, leaving the door open for a future
improvement:

https://github.com/tc39/test262/pull/191#issuecomment-77442739